panther_manager/include/panther_manager/plugins/shutdown_hosts_node.hpp

```diff
@@ -0,0 +1,177 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_MANAGER_SHUTDOWN_HOSTS_NODE_HPP_
+#define PANTHER_MANAGER_SHUTDOWN_HOSTS_NODE_HPP_
+
+#include <algorithm>
+#include <memory>
+#include <numeric>
+#include <set>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <behaviortree_cpp/action_node.h>
+#include <behaviortree_cpp/basic_types.h>
+#include <behaviortree_cpp/tree_node.h>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <panther_manager/plugins/shutdown_host.hpp>
+
+namespace panther_manager
+{
+
+class ShutdownHosts : public BT::StatefulActionNode
+{
+public:
+  explicit ShutdownHosts(const std::string & name, const BT::NodeConfig & conf)
+  : BT::StatefulActionNode(name, conf)
+  {
+    // TODO: @delihus What is the name of ros::this_node?
+    node_name_ = name;
+    logger_ = std::make_shared<rclcpp::Logger>(rclcpp::get_logger(node_name_));
```

Comment:
not sure if this can be done without passing node to plugin. In my opinion the best we can do is use BT node name, as you did, then node_name_ can be removed
```suggestion
    logger_ = std::make_shared<rclcpp::Logger>(rclcpp::get_logger(name));
```